src/services/infra/InfraService.ts

```diff
@@ -193,17 +197,18 @@ export default class InfraService {
        * amplify will store the prefix as "www". To get the entire redirectionDomainSource,
        * I would have to add the prefix ("www") with the primary domain (blah.gov.sg)
        */
-      const redirectionDomainSource = `${redirectionDomainList?.[0].subDomainSetting?.prefix}.${primaryDomain}`
-
       const userId = agency.id
-      const newLaunchParams = {
+      const newLaunchParams: launchesCreateParamsType = {
         userId,
         siteId,
         primaryDomainSource: primaryDomain,
         primaryDomainTarget,
         domainValidationSource,
         domainValidationTarget,
-        redirectionDomainSource,
+      }
+
+      if (redirectionDomainList?.length) {
+        newLaunchParams.redirectionDomainSource = `${redirectionDomainList[0].subDomainSetting?.prefix}.${primaryDomain}`
```

Comment:
could i get some context on why we're using the [0]th index element? this would be `www` right? if so, i think it'd be clearer to just have `www.${primaryDomain}`

Reply:
thats fair - 3773fbf